File: 450 DSA/Arrays/Kadane's algo/Kadane's algo.py
a = [74,-72,94,-53,-59,-3,-66,36,-13,22,73,15,-52,75]

def maxSubArraySum(a):
    current_sequence = 0
    best_sequence = a[0] # Assigning first number of array to best_sequence
    for x in a:
        current_sequence = max(x,current_sequence+x) # Finding the maximum between the current number and sum of current number and sequence
        best_sequence = max(best_sequence,current_sequence)
    return best_sequence     # Returns the sequence whose sum is the highest

# Print the array
print("Largest sum contiguous subarray:",maxSubArraySum(a))

File: 450 DSA/Arrays/Kadane's algo/Kadane's algo.java
// Java program to print largest contiguous array sum
import java.io.*;
import java.util.*;
 

class KadaneAlgorithm
{
    public static void main (String[] args)
    {
        //given Input array
        int [] a = {74,-72,94,-53,-59,-3,-66,36,-13,22,73,15,-52,75};
        System.out.println("Maximum contiguous sum is " +
                                       maxSubArraySum(a));
    }
 
    static int maxSubArraySum(int a[])
    {
        int size = a.length;
        // max calculate the maximum sum so far
        // sum calculate the running sum of array
        int max = Integer.MIN_VALUE, sum = 0;
 
        for (int i = 0; i < size; i++)
        {
            sum = sum + a[i];
            //if running sum is greater than maximum sum update it
            if (max <sum)
                max = sum;
            // if sum is -ve than start for new subarray with sum=0
            if (sum < 0)
                sum = 0;
        }
        return max;
    }
}

// output for above input=156
// Time complexity O(n)